### DISCUSSION TITLE
[FEATURE REQUEST] Split grains, pillar, and mine caches

### DISCUSSION THREAD

**Discussion by nf-brentsaner** [saltstack/salt#67284] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Currently, `minion_data_cache` via master configs only supports a boolean. There is no way to explicitly enable/disable the grain cache, pillar cache, and/or mine cache.

Further, if one wishes to disable caching of these, mines are rendered completely nonfunctional. (Which I guess makes sense/I understand why, now, but the configuration comment is nowhere near clear about this and if one wishes to disable caching of grains or the pillar, they must also completely sacrifice mine functionality.)

**Describe the solution you'd like**
Configuration directives to selectively enable/disable caches per grains/pillar/mine. (And, for bonus points, the ability to selectively do this per minion targeting as well.)

**Describe alternatives you've considered**
(There are none)

**Additional context**
I believe they already cache to separate files (`<master cache>/minions/<minion id>/{data,pillar,mine}.p`)?. If so, this should have minimal impact.

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by whytewolf** [comment]
so pillar cache does have it's own setting. and it isn't enabled by default. see https://docs.saltproject.io/en/latest/ref/configuration/master.html#pillar-cache-options

**Comment by nf-brentsaner** [comment]
Oh! Thanks, I missed that! I was checking the stock config file itself from 3006 and didn't see it commented there.
Is it possible to split out the mine cache as well like that?
